extension/src/paymentHandler/payment-utils.js

```diff
@@ -133,6 +133,30 @@ function createAddTransactionAction ({
   }
 }
 
+function createAddTransactionActionByResponse (amount, currencyCode, response) {
+  // eslint-disable-next-line default-case
+  switch (response.resultCode) {
+    case 'Authorised':
+      return createAddTransactionAction({
+        type: 'Authorization',
+        state: 'Success',
+        amount,
+        currency: currencyCode,
+        interactionId: response.pspReference
+      })
+    case 'Refused':
```

Comment:
for Refused and Error, we are simply using the create new payment object practice, do you think some additional edge cases for this, maybe if it's a retyable according to `refusalReason` field maybe we should not add a transaction? Should we ask that to adyen, WDYT? 

check: https://docs.adyen.com/development-resources/refusal-reasons

Reply:
I think we already discussed this that we should not retry? 

However, when I look at the response codes, I guess in some cases it's doable. But the retry could also be done by the shop by creating a new payment, so I'm not sure if it make sense here. Also lots of reasons they have as Retryable is not retryable for us (e.g. retry with new payment method).

Reply:
We discussed both not in detail, I agree it does not make sense in here. We could open this as an issue and return later after clarification, maybe we could add it as a todo with GitHub issue link.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/issues/276